specs/typography/heading.md

```diff
@@ -0,0 +1,241 @@
+2. Defining accessible page hierarchy through one or more sequences of ```<h1>```, ```<h2>```, ```<h3>```, etc. elements
+3. Formatting heading elements off-ramp, i.e. a smaller ```<h2>``` that groups a section and larger ```<h3>``` for main topics within that section
+ 
+### Features
+
+1. Ability to use any size in the type ramp as a heading
+2. A default mapping of systematic styles to standard elements
+3. Use of a contiguous ramp sequence to create intentional rhythm
+4. A set of semantically named styles based on the underlying type ramp
+
+### Risks and Challenges
+
+It's possible that styles alone (not components) will be harder to implement consistently. We may want to consider simple custom elements that consistently apply the same underlying styles as well as any necessary aria tags.
+
+### Prior Art / Examples
+
+- FAST-DNA (React): [Heading](https://explore.fast.design/components/heading), [Subeading](https://explore.fast.design/components/subheading)
+- [Material Design](https://material.io/design/typography/the-type-system.html)
+- [Bootstrap](https://getbootstrap.com/docs/4.4/content/typography)
+- [Ant Design](https://ant.design/components/typography)
+- [Mineral UI](https://mineral-ui.netlify.app/typography#display-text)
+- [Semantic UI](https://semantic-ui.com/elements/header.html)
+- [Fluent UI](https://fluentsite.z22.web.core.windows.net/components/header/definition)
+
+---
+
+## Design
+
+A heading can be made from any size in the type ramp.
+
+![](./images/heading-examples.png)
+
+### API
+
+Pseudo interfaces: (*Could be functions, decorators, design system properties, or otherwise*)
+
+#### Basic heading style map
+
+Bare-bones styles applied manually.
+
+```TypeScript
+headingStyles.basicRamp()
+```
+
+Produces css classes that can be applied to any text elements. The classes are consistently named for relation to the raw type ramp.
+
+*Supports use cases 1 & 3, feature 1*
+
+```CSS
+.fast-heading-minus-2 ...
+
+.fast-heading-minus-1 ...
+
+.fast-heading-base {
+    font-size: var(--fast-type-ramp-base-size);
+    line-height: var(--fast-type-ramp-base-height);
+    font-weight: var(--fast-heading-font-weight);
+}
+
+.fast-heading-plus-1 ...
+
+...
+```
+
+Usage:
+
+```HTML
+<h1 class="fast-heading-plus-3">If all the world is a stage, I want better lighting</h1>
+
+<span class="fast-heading-minus-2">If all the world is a stage, I want better lighting</span>
+```
+
+![](./images/basic-heading-samples.png)
+
+#### Scoped standard heading element map
+
+The easiest option for accessible content when you know how many levels you have.
+
+```TypeScript
+headingStyles.standardElementMap(levels: 3, includeBase: false)
+```
+
+Applies the styles to the standard heading elements for the desired number of levels so the adaptive styling is used without needing to add and maintain class names. Starts directly above the base type ramp size or optionally includes that size depending on stylistic preference.
```

Comment:
Can you elaborate on how these would apply to standard headings? Is this a utility rather than a component? It seems like this is applying values directly to standard elements, but with our current implementations it seems like this would be scoped to within the Shadow DOM and I can't quite think of a place where we would hard code a heading level in our components.

Reply:
Yes, I think under the current idea of using standard markup it's more of a utility than a component. I believe this is the direction we were considering for elements like this where there wasn't much else being added by a component.

Per what I recalled from a discussion a while back, we could set styles and let them chain down the same way we're doing with the css variables that represent the design system properties. That is what I was thinking here, that whatever the final interface is for this, you would do it in conjunction with your root design system setup. I wasn't exactly sure what that would look like, which is why I commented these as pseudo interfaces.

Let's setup a quick chat if my assumptions are incorrect.

Reply:
Yes, I don't think these are components, so that assumption is correct. As a utility I assume we're providing a methodology that someone could use to create generate this. Could this be configured to use classes if I wanted instead of the standard element?

Reply:
Yes, you could use `basicRamp` or `simplePageStyleMap` or write your own use case specific version of the latter.